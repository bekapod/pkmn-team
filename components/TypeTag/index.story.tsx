import { storiesOf } from "@storybook/react";
import React from "react";
import TypeTag from ".";

storiesOf("TypeTag", module).add(
  "default",
  (): JSX.Element => <TypeTag type="electric">Electric</TypeTag>
);
