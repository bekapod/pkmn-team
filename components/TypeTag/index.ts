import styled from "styled-components/macro";
import { getTypeColor } from "../../helpers/general";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";
import { Type } from "../../types"; // eslint-disable-line import/named

interface Props {
  type: Type;
}

const TypeTag = styled.span`
  padding: 0 ${variables.spacing.xs}px;
  color: ${variables.colors.white};
  font-size: ${variables.fontSizes.xs}px;
  font-weight: 700;
  line-height: ${lineHeight("xs")};
  text-shadow: 0 1px 0px ${variables.colors.grayDarker},
    1px 0 0px ${variables.colors.grayDark};
  text-transform: uppercase;
  background-color: ${({ type }: Props) => getTypeColor(type)};
  border-radius: ${variables.sizes.borderRadiusSm}px 0;
`;

export default TypeTag;
