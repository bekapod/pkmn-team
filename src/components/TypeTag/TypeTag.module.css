.base {
  display: inline-block;
  padding: 0 var(--spacing-2);
  color: var(--color-white);
  font-size: var(--font-size-sm);
  font-weight: 700;
  line-height: var(--line-height-sm);
  text-shadow: 0 1px 0px var(--color-sherpa-blue-900),
    1px 0 0px var(--color-sherpa-blue-800);
  text-transform: uppercase;
  background-color: var(--type-color);
  border-radius: var(--border-radius-sm) 0;
}
