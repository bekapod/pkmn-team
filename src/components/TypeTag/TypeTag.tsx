import styled from 'styled-components/macro';
import { getTypeColor } from '~/lib/general';

export type TypeTagProps = {
  type: string;
};

export const TypeTag = styled.span<TypeTagProps>`
  display: inline-block;
  padding: 0 var(--spacing-xs);
  color: var(--color-white);
  font-size: var(--font-size-xs);
  font-weight: 700;
  line-height: var(--spacing-md);
  text-shadow: 0 1px 0px var(--color-gray-darker),
    1px 0 0px var(--color-gray-dark);
  text-transform: uppercase;
  background-color: ${({ type }) => getTypeColor(type)};
  border-radius: var(--border-radius-sm) 0;
`;
