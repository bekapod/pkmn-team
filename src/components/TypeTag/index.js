// @flow
import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";
import { getTypeColor } from "../../helpers/general";
import type { Type } from "../../types";

type Props = {
  type: Type
};

const TypeTag = styled.span`
  padding: 0 ${variables.spacing.xs}px;
  color: ${variables.colors.white};
  font-size: ${variables.fontSizes.xs}px;
  font-weight: 700;
  line-height: ${lineHeight("xs")};
  text-shadow: 0 1px 0px ${variables.colors.grayDarker},
    1px 0 0px ${variables.colors.grayDark};
  text-transform: uppercase;
  background-color: ${({ type }: Props) => getTypeColor(type)};
  border-radius: ${variables.sizes.borderRadiusSm}px 0;
`;

export default TypeTag;
