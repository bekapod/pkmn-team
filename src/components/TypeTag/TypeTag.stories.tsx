import { Meta, Story } from '@storybook/react/types-6-0';
import { ComponentProps } from 'react';
import { TypeTag } from './TypeTag';

export default {
  title: 'Components/Type Tag',
  component: TypeTag,
  args: {
    typeSlug: 'electric'
  }
} as Meta<ComponentProps<typeof TypeTag>>;

export const typeTag: Story<ComponentProps<typeof TypeTag>> = args => (
  <TypeTag {...args}>Type name</TypeTag>
);
