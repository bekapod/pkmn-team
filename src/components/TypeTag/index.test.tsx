import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import TypeTag from "./";

describe("<TypeTag />", () => {
  it("renders without crashing with ELECTRIC type", () => {
    const tree = renderer.create(<TypeTag type="ELECTRIC">ELECTRIC</TypeTag>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders without crashing with PSYCHIC type", () => {
    const tree = renderer.create(<TypeTag type="PSYCHIC">PSYCHIC</TypeTag>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
