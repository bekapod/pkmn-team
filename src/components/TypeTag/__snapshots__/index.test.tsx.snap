// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeTag /> renders without crashing with ELECTRIC type 1`] = `
<span
  className="TypeTag-sc-1dttu8y-0 byzsso"
  type="ELECTRIC"
>
  ELECTRIC
</span>
`;

exports[`<TypeTag /> renders without crashing with PSYCHIC type 1`] = `
<span
  className="TypeTag-sc-1dttu8y-0 hSnHTZ"
  type="PSYCHIC"
>
  PSYCHIC
</span>
`;
